File: Seminar_05/Model/ComplexModel/SumComplex.java
package Seminar_05.Model.ComplexModel;

import Seminar_05.Model.CalcModel;

public class SumComplex extends CalcModel {

    @Override
    public double result(double x, double y) {
        return x + y;
    }

    public Complex resultComplex(double x1, double x2, double y1, double y2){
        double newReal = result(x1, x2);
        double newImage = result(y1, y2);
		Complex result = new Complex(newReal,newImage);
		return result;  
    }
}

File: Seminar_07/Model/Complex_model/SumComplex.py
import Seminar_07.Model.Rational_model.SumRational as Sum
import Seminar_07.Model.Complex_model.ComplexNumber as ComplexNumber

class SumComplex():

    def result(self, x1, x2, y1, y2):
        new_real = Sum.SumRational.result(self, x1, x2)
        new_image = Sum.SumRational.result(self, y1, y2)
        result = ComplexNumber.Complex(new_real, new_image)
        return result